lib/web-service.js

```diff
@@ -203,14 +203,28 @@ export const addLicenseFilePath = async (packagesInfos, httpRetryOptions = {}, a
         const githubPackageName = getPackageNameFromLink(element.link);
         if (githubPackageName.length !== 0) {
           const licenseInfoAdr = `https://api.github.com/repos/${githubPackageName}/license`;
+          let options = {};
           try {
-            const options = gotGithubJsonOptions(httpRetryOptions.maxAttempts, githubPackageName, authorizationOptions);
+            options = gotGithubJsonOptions(httpRetryOptions.maxAttempts, githubPackageName, authorizationOptions);
             const result = await got(licenseInfoAdr, options);
             const licenseInfo = result.body;
             if ((licenseInfo.download_url !== undefined) && (licenseInfo.download_url.length > 0)) {
               element.licenseFileLink = licenseInfo.download_url;
             }
           } catch (error) {
+            const { response } = error;
+            if (response?.statusCode === 403) {
+              const { headers } = response;
+              if (headers['x-ratelimit-remaining'] === '0') {
+                const timeForRateLimitReset = parseInt(error.response.headers['x-ratelimit-reset']) - Math.floor(new Date().getTime() / 1000);
+                const resetMsg = timeForRateLimitReset ? ` please wait ${timeForRateLimitReset} seconds before trying again.` : '';
+                console.error(`${githubPackageName}: License query failed. Rate limit of ${headers['x-ratelimit-limit']} requests per hour exceeded.${resetMsg}`);
+                if (!options.headers?.Authorization)
+                  console.error('You can increase the rate limit by providing a github token. See the README for more information.');
```

Comment:
> ... README ...

While we are at it: shouldn't we mention 'fine-grained token' in the README (line 47) and reference [Get the license for a repository](https://docs.github.com/en/rest/licenses/licenses?apiVersion=2022-11-28#get-the-license-for-a-repository) (with or better without the link which will change at any time)?

Reply:
Sure, but these are out of scope for this PR.